frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -396,12 +529,14 @@ impl<T: Config> Pallet<T> {
 	/// not.
 	///
 	/// This essentially makes sure that we don't run on previous blocks in case of a re-org, and we
-	/// don't run twice within a window of length [`OFFCHAIN_REPEAT`].
+	/// don't run twice within a window of length `threshold`.
 	///
 	/// Returns `Ok(())` if offchain worker should happen, `Err(reason)` otherwise.
-	pub(crate) fn try_acquire_offchain_lock(now: T::BlockNumber) -> Result<(), &'static str> {
-		let storage = StorageValueRef::persistent(&OFFCHAIN_HEAD_DB);
-		let threshold = T::BlockNumber::from(OFFCHAIN_REPEAT);
+	pub(crate) fn try_acquire_offchain_lock(
+		now: T::BlockNumber,
+	) -> Result<(), MinerError> {
+		let threshold = T::OffchainRepeat::get();
+		let storage = StorageValueRef::persistent(&OFFCHAIN_LOCK);
 
 		let mutate_stat =
 			storage.mutate::<_, &'static str, _>(|maybe_head: Option<Option<T::BlockNumber>>| {
```

Comment:
sidenote about the code below:

As far as I know, we run offchain worker only on the best block, if we do reorg we don't run the offchain worker on every block in the reorg.

If this is true then we could run the offchain worker when `now` is less than `head` on forks, because if we consider another chain to be the best then we should submit to this chain, no matter what we did in a previous fork. no ?